spec/features/applicant_page_spec.rb

```diff
@@ -154,60 +119,17 @@
       visit_applicant_page
       select_applicant_boolean(:over_60, false)
       select_applicant_boolean(:employed, false)
+      select_applicant_boolean(:passporting, false)
+      click_on "Save and continue"
     end
 
-    describe "passporting" do
-      before do
-        select_applicant_boolean(:dependants, false)
-        select_applicant_boolean(:passporting, passporting)
-      end
-
-      context "when passporting" do
-        let(:passporting) { true }
-
-        before do
-          click_on "Save and continue"
-        end
-
-        it "skips income and outgoings" do
-          expect(page).to have_content property_header
-        end
-
-        it "has a back pointer to the applicant page" do
-          click_on "Back"
-          expect(page).to have_content applicant_header
-        end
-      end
-
-      context "without passporting" do
-        let(:passporting) { false }
-
-        before do
-          click_on "Save and continue"
-        end
-
-        it "shows income" do
-          expect(page).to have_content benefits_header
-        end
-
-        it "has a back pointer to the applicant page" do
-          click_on "Back"
-          expect(page).to have_content applicant_header
-        end
-      end
+    it "goes on to dependants screen" do
+      expect(page).to have_content dependants_header
     end
 
-    describe "dependants field" do
-      before do
-        select_applicant_boolean(:dependants, true)
-        select_applicant_boolean(:passporting, false)
-      end
-
-      it "submits dependants" do
-        fill_in "applicant-form-dependant-count-field", with: "2"
-        click_on "Save and continue"
-        expect(page).to have_content benefits_header
-      end
+    it "has a back pointer to the applicant page" do
+      click_on "Back"
+      expect(page).to have_content applicant_header
```

Comment:
In the past for matching e.g. page titles I have used the following format:

expect(page.body).to include(I18n.t("estimate_flow.dependants.legend"))

This means you don't need the let at the top of the page. Happy to learn that there is a reason not to do it this way also.

Reply:
IMHO it's useful to give the thing you're checking for a name - we actually want to say 'we are on page x' and there are many ways of doing that. Originally I started checking the URL, but I felt like it might be more fragile than checking for some basic header content.